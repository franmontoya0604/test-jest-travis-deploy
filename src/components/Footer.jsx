import React from 'react'
import './styles/footer.css'


function Footer() {
    return (
        <div className="bg-secondary Footer container-fluid mt-5">
           
            <h2 className="Footer-title">proyecto hooks</h2>
            
            <h2 className="Footer-copy">derechos reservados de fran</h2>
           
        </div>
    )
}

export default Footer
