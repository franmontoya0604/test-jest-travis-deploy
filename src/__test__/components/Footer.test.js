import React from "react";
import {shallow} from 'enzyme';
import Footer from '../../components/Footer';
import renderer from 'react-test-renderer'; // de este utils usamoscreate para renderizar.

///lo mejor es usar act cuando se renderiza el componente en un dom creado. ya que act asegura que antes de hacer el expect el renderizado qhay terminado.
describe("render de componentes",() => {  //en el curso usan describe yo uso it, pero no puedo por que test van con describe y act con it.
          const footer =  shallow(<Footer/>) ; //asi guardamo lo que devuelve el renderizado. en la documentacion de react hacen el renderizado en uncontainer que es un div creado en el momento. ver cual es mejor forma.

    test("render de footer",()=> {  //en el curso usna test , yo voy a usar act ,cuando usamos act el expect debe estar afuera y aca no puede estar afuera por que el render se hace en una variable que el scope es en la funcion nomas, por lo tanto tengo que usar test
    expect(footer.length).toEqual(1);  //aca decimos la cantidad que hay en el footer osea 1.
    })

    test("render del tittle",() => {
        expect(footer.find(".Footer-title").text()).toEqual("proyecto hooks");  //aca lo que hacemos es usar find para buscar el elemento atraves de su class y ver si su texto es igual al toEqual.
    })

},


describe('footer snapshot', () => {
    test('Comprobar la UI del componente Footer', () => {
      const footer = renderer.create(<Footer />).toJSON(); //aca renderizamos el footer con el create y lo transformamos  en objeto json.
  
      expect(footer).toMatchSnapshot(); // y aca comprobamos el snapshot y si es la primera ves se crea.
    });
  })

)