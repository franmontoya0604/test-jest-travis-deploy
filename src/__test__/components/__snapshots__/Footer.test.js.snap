// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer snapshot Comprobar la UI del componente Footer 1`] = `
<div
  className="bg-secondary Footer container-fluid mt-5"
>
  <h2
    className="Footer-title"
  >
    proyecto hooks
  </h2>
  <h2
    className="Footer-copy"
  >
    derechos reservados de fran
  </h2>
</div>
`;
